br/pkg/lightning/restore/check_info.go

```diff
@@ -598,15 +599,16 @@ func hasDefault(col *model.ColumnInfo) bool {
 		col.IsGenerated() || mysql.HasAutoIncrementFlag(col.Flag)
 }
 
-func (rc *Controller) readFirstRow(ctx context.Context, dataFileMeta mydump.SourceFileMeta) (cols []string, row []types.Datum, err error) {
+func (rc *Controller) createDataFileParser(ctx context.Context, dataFileMeta mydump.SourceFileMeta, replaceStr string) (mydump.Parser, error) {
```

Comment:
`replaceStr` is not clear what it means. I think we should update it to a better name.

Reply:
what do you suggest

Reply:
`dataInvalidCharReplace` would be ok. It's the same with the argument of `NewCharsetConvertor` and looks more clear.